src/stores/RoomListStore.js

```diff
@@ -68,6 +78,15 @@ class RoomListStore extends Store {
                 this._generateRoomLists();
             }
             break;
+            case 'MatrixActions.Room.timeline': {
+                if (!this._state.ready ||
+                    !payload.isLiveEvent ||
+                    !payload.isLiveUnfilteredRoomTimelineEvent ||
+                    !this._eventTriggersRecentReorder(payload.event)
+                ) break;
+                this._generateRoomLists();
```

Comment:
shouldn't _generateRoomLists have a ratelimiter on it?  surely this is quite a heavy operation, and we shouldn't be doing it each and every time a live timeline event happens?